### PR TITLE
Clean up branching errors

### PR DESCRIPTION
**Pull request by vernondcole**
I fouled things up by committing directly to the master branch. I have merged all changes into the develop branch -- this PR is to bring the merged fixes back to master.
  Also in this set is a skeleton of a cloud provider for Azure. It wants more support software loaded in order to use it, so I added an "azure:enabled" pillar value to turn the whole thing on or off.

### CODE REVIEW

File: Vagrantfile [hunk 1]
```diff
@@ -164,10 +164,10 @@ Vagrant.configure(2) do |config|  # the literal "2" is required.
...
       end
     end
-    if vagrant_command == "ssh"
-      master_config.ssh.username = my_linux_user  # if you type "vagrant ssh", use this username
-      master_config.ssh.private_key_path = Dir.home() + "/.ssh/id_rsa"
-    end
+    #if vagrant_command == "ssh"
+    #  master_config.ssh.username = my_linux_user  # if you type "vagrant ssh", use this username
+    #  master_config.ssh.private_key_path = Dir.home() + "/.ssh/id_rsa"
+    #end
```
Comment (tedjp) [inline Vagrantfile:169]: Why's this commented-out? Should it be deleted?

File: Vagrantfile [hunk 2]
```diff
@@ -345,8 +346,9 @@ Vagrant.configure(2) do |config|  # the literal "2" is required.
  # . . . . . . . . . . . . Define machine win16 . . . . . . . . . . . . . .
  # . this machine installs Salt on a Windows 2016 Server and runs highstate.
   config.vm.define "win16", autostart: false do |quail_config|
-    quail_config.vm.box = "gusztavvargadr/w16s" # Windows Server 2016 standard
-    quail_config.vm.box_version = "1808.0.0" # WARNING: 1809.0.0 does not have WinRM enabled
+    quail_config.vm.box = "cdaf/WindowsServer" #gusztavvargadr/w16s" # Windows Server 2016 standard
+    # "jacqinthebox/windowsserver2016"
+    # quail_config.vm.box_version = "1808.0.0" # WARNING: 1809.0.0 does not have WinRM enabled
```
Comment (tedjp) [inline Vagrantfile]: Should these comments be deleted too?

**Review by tedjp — APPROVED** [review]

File: Vagrantfile (hunk 1 above)
Comment (vernondcole) [reply Vagrantfile:169]: I left this in on purpose, as an example of what someone might want to do.  I thought this was really cool when I put it in, but it has ended up being a foot gun too often, so I commented it and will use a normal "ssh -A bevymaster" when I want to log in using my own username, and "vagrant ssh bevymaster" will log in using the default "vagrant" username.

File: Vagrantfile (hunk 2 above)
Comment (vernondcole) [reply Vagrantfile]: I was lazy. These are other VM package that I have tried and though I might want to revert to.
I'll pull the comments now and let it come through on the next revision.

**Review by goutkannan — APPROVED** [review]
For our project, it would be helpful if we have a dev win server with a visual studio community edition installed. 
Say, use choco to install windows dev tools.

**Pull request merged**